src/composables/trade/useJoinExit.spec.ts

```diff
@@ -0,0 +1,114 @@
+import { parseFixed } from '@ethersproject/bignumber';
+import { computed, ref } from 'vue';
+import { mount } from '@/tests/mount-composable-tester';
+
+import useJoinExit from '@/composables/trade/useJoinExit';
+
+jest.mock('vue-i18n');
+jest.mock('vuex');
+jest.mock('@/composables/useEthereumTxType');
+jest.mock('@/composables/useEthers');
+jest.mock('@/composables/useTransactions');
+jest.mock('@/locales');
+jest.mock('@/services/web3/useWeb3');
+jest.mock('@/services/rpc-provider/rpc-provider.service');
+jest.mock('@/composables/queries/useRelayerApprovalQuery');
+
+jest.mock('@/providers/tokens.provider', () => ({
+  useTokens: () => {
+    return {
+      injectTokens: jest.fn().mockImplementation(),
+      priceFor: jest.fn().mockImplementation(),
+      useTokens: jest.fn().mockImplementation(),
+      getToken: jest.fn().mockImplementation(),
+    };
+  },
+}));
+
+jest.mock('@/composables/approvals/useRelayerApproval', () => ({
+  __esModule: true,
+  default: () =>
+    jest.fn().mockImplementation(() => ({
+      relayerSignature: '-',
+    })),
+  RelayerType: {
+    BATCH_V4: 'BATCH_V4',
+  },
+}));
```

Comment:
Our current code forces to do a very complex mocking setup. We will simplify this in incoming PRs.